### PR TITLE
First opentelemetry integration trial package for research purpose.

### PR DESCRIPTION
**Pull request by lalit-satapathy**
## What does this PR do?

**Exploratory package** to post-process opentelemetry data from apm-server. Neither it is for end-user consumption, nor is it a representative otel integration. For **internal research & review** purpose only.

Uses a **dummy data stream** (input http/metrics) for apm-server generated document post-processing.

CC: @ruflin @rameshelastic @andresrc @pmeresanu85 

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] PR for illustrative purpose only and no plan to merge this now.

## How to test this PR locally

- Run stack
- Run apm-server
- Run otel collector with correct service.name
- Install integration

## Related issues

- Relates https://github.com/elastic/observability-dev/issues/2293 
- Relates https://github.com/elastic/observability-dev/issues/2460

## Screenshots

Screenshot 2022-12-07 at 10 37 31 AM

## Future
- Field mapping

### CODE REVIEW

File: packages/otel/data_stream/dummy/agent/stream/stream.yml.hbs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+metricsets: ["json"]
```
Comment (ruflin) [inline packages/otel/data_stream/dummy/agent/stream/stream.yml.hbs:1]: Thinking about otel configs for apm-server: What if there would be an input called `otel-receiver`:

```
  - type: apm
    apm-server:
      host: 'localhost:8200'
      ...
  - type: otel-receiver
    routing:
		a.b == c
		data_stream.dataset = foo
```
Any otel config would also be shipped to the apm-server and would then make decision on where data should be shipped. Conditions above or only as example, not valid yaml ...

@axw Just some ideas.

File: packages/otel/_dev/build/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# OpenTelemetry Package
+
+Exploratory package to post-process opentelemetry data from apm-server. Neither it is for end-user consumption, nor is it a representative otel integration. For internal research & review purprose only.
+
+Processes opentelemetry documents generated by [apm-server](https://www.elastic.co/guide/en/apm/guide/current/open-telemetry.html) using opentelemetry [collector](https://opentelemetry.io/docs/collector/) for [hostmetrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/hostmetricsreceiver/README.md).
+
+## Assumptions
+* opentelemetry-collector to provide *service.name* as dummy. The dataset binding is based on it.
+* Uses a dummy data stream for now, which just pings(localhost:9200") as a redundant step.
+* Uses hostmetrics system.filesystem mappings in ingest pipeline mapping.
```
Comment (ruflin) [inline packages/otel/_dev/build/docs/README.md:10]: I wonder if a better example would be to take postgres, mysql or similar data to see how it fits.

File: packages/otel/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+format_version: 1.0.0
+name: otel
+title: "OpenTelemetry Package"
+version: 0.1.0
+license: basic
+description: "Exploratory Package to process otel data from apm-server."
+type: integration
```
Comment (ruflin) [inline packages/otel/manifest.yml:7]: Are you building the input package or an integration package for a specific service with otel?

Comment (lalit-satapathy) [reply packages/otel/manifest.yml:7]: This is experimental integration verifies(proof-of-concept) post-processing of apm-server generated otel documents using ingest pipeline. 

We will need to explore both a) Input package b) Integrations for different usages of OTEL, possibly for different use cases.

**Comment by lalit-satapathy** [comment]
Removed redundant _policy_templates inputs_ section, so that the open-telemetry package can be installed without any additional configuration. 
Screenshot 2022-12-29 at 10 47 18 AM

**Pull request closed without merging**